cmd/frontend/graphqlbackend/external_service_collaborators.go

```diff
@@ -0,0 +1,124 @@
+package graphqlbackend
+
+import (
+	"context"
+	"net/url"
+	"sort"
+	"strings"
+	"sync"
+	"time"
+
+	"github.com/cockroachdb/errors"
+	"github.com/inconshreveable/log15"
+	"github.com/sourcegraph/sourcegraph/internal/extsvc/auth"
+	"github.com/sourcegraph/sourcegraph/internal/extsvc/github"
+	"github.com/sourcegraph/sourcegraph/internal/goroutine"
+	"github.com/sourcegraph/sourcegraph/schema"
+)
+
+// See schema.graphql for an explanation of how this is intended to be used. This is particularly
+// for listing collaborators to *some* of the repositories associated with this external service
+// *before* they are cloned onto Sourcegraph.
+func (r *externalServiceResolver) InvitableCollaborators(ctx context.Context) ([]*invitableCollaboratorResolver, error) {
+	cfg, err := r.externalService.Configuration()
+	if err != nil {
+		return nil, errors.Wrap(err, "failed to parse external service config")
+	}
+	githubCfg, ok := cfg.(*schema.GitHubConnection)
+	if !ok {
+		return nil, errors.Wrap(err, "contributors API only supported with GitHub")
+	}
+	githubUrl, err := url.Parse(githubCfg.Url)
```

Comment:
Don't we need to do this so it also works properly for github.com?

```suggestion
    baseURL, err := url.Parse(githubCfg.Url)
	if err != nil {
		return nil, errors.Wrap(err, "failed to parse external service URL")
	}
	baseURL = extsvc.NormalizeBaseURL(baseURL)

	githubUrl, _ := github.APIRoot(baseURL)
```